tempodb/backend/s3/s3.go

```diff
@@ -138,9 +144,7 @@ func (rw *readerWriter) Append(ctx context.Context, name string, keypath backend
 	var a appendTracker
 	objectName := backend.ObjectFileName(keypath, name)
 
-	options := minio.PutObjectOptions{
-		PartSize: rw.cfg.PartSize,
-	}
+	options := getTempoObjectOptions(rw)
```

Comment:
This does add UserTags here which feels like a bug fix. Can you think of any reason we wouldn't want that?

Reply:
Thought about it a little. 
This will just set the user tags to the new user tags during append. 
Sounds to me like something that leads to a negligible performance impact and will only lead to a behavior change if someone changes the user tags (which in this case will lead to the new user tags when appending) 

Seems to me that if anything this is a good thing unless there is a performance change. Do you think that this could lead to a performance change? If so I'll be more than happy to change it

Reply:
Agree it seems like a great bug fix.  It doesn't look like the headers are supported on [UploadPart](https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html) but they are supported on [CreateMultipartUpload](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateMultipartUpload.html).  Since it's working then either the extra headers don't matter or the minio client is handling it.   Have you ran this against real S3 and seen any issues?

Reply:
Haven't ran it against a real S3. Feel free to check and I'll make any changes necessary :)